### PR TITLE
Supports built-in WebP/AWebP codec from ImageIO for iOS 14/tvOS 14/macOS 11/watchOS 7

### LINKED ISSUE #3041: Feature: Supports WebP/Animated WebP nativally for iOS 14/macOS 11/tvOS 14/watchOS 7 without extra libwebp dependency
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none
### Issue Description and Steps

Image/IO in iOS 14/macOS 14/tvOS 11/watchOS 7 supports WebP (including Animated WebP) nativally. See full changelog: http://codeworkshop.net/objc-diff/sdkdiffs/ios/14.0/ImageIO.html

It's easy and right time to drop the libwebp for user who set minimun deployment target version to iOS 14+.

ImageIO use hardware-accelerated codec instead of that libwebp (which is written in C and run on CPU), is much faster than the current WebP codecs on iOS 13.

### Design

Introduce the new subclass for AWebP: `SDImageAWebPCoder`. The `SDImageWebPCoder` naming already been taken by another codec repo maintained by us. So this a new naming.

In the future (2 Years later ?), that [SDWebImageWebPCoder](https://github.com/SDWebImage/SDWebImageWebPCoder) repo may be deprecated.

Note: For Static WebP, you don't need to add `SDImageAWebPCoder`, the `SDImageIOCoder` already handle those cases.

### Usage

For iOS 14+ user, you can use avaiable check to use the native WebP support. For iOS 13-, use the fallback version of libwebp support.

```objective-c
if (@available(iOS 14, macOS 11, tvOS 14, watchOS 7, *)) {
    [SDImageCodersManager.sharedManager addCoder:SDImageAWebPCoder.sharedCoder];
} else {
    [SDImageCodersManager.sharedManager addCoder:SDImageWebPCoder.sharedCoder];
}
```

### PR DESCRIPTION
**Pull request by dreampiggy**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: #3041 

### Pull Request Description

This close #3041

### Changes

Similiar to HEIC, iOS 14's new WebP codec can be implementes with the subclass of `SDAnimatedImageIOCoder`.

~~The `kUTTypeWebP` does not get defined by Apple's SDK (bugs?), but I found the raw string value for that is `org.webmproject.webp`, so we define one with macro~~

The UTI API is here: https://developer.apple.com/documentation/uniformtypeidentifiers/uttypewebp
However, we need to support Xcode 11 :) So we hardcode the raw value with macro.

For macOS, we can supports the NSImageView rendering with `SDAnimatedImageRep` as well.

Also, this PR changes one possible previouslly wrong design, now, the `SDImageIOCoder` as a default backup, does not check any format and always pass it to system and try to decode, no need hack for `HEIC`, or `WebP`.

Run result on iPhone Simulator with iOS 14:

image

### CODE REVIEW

File: Examples/SDWebImage Demo.xcodeproj/xcshareddata/xcschemes/SDWebImage Watch Demo.xcscheme [hunk 1]
```diff
@@ -112,7 +90,7 @@
       debugDocumentVersioning = "YES">
       <RemoteRunnable
          runnableDebuggingMode = "2"
-         BundleIdentifier = "com.apple.carousel"
+         BundleIdentifier = "com.apple.Carousel"
```
Comment (dreampiggy) [inline Examples/SDWebImage Demo.xcodeproj/xcshareddata/xcschemes/SDWebImage Watch Demo.xcscheme:93]: This diff is caused by Xcode 12 update. And seems does not effect the user who use Xcode 11.

**Comment by dreampiggy** [comment]
During test, seems Image/IO support does not supports encoding.

This is limited by Image/IO, hope future we can have built-in WebP encoding as well.

image

**Comment by dreampiggy** [comment]
Fired Radar: FB7844481

image

File: SDWebImage/Core/SDImageAWebPCoder.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#import <Foundation/Foundation.h>
+#import "SDImageIOAnimatedCoder.h"
+
+/**
+ This coder is used for Google WebP and Animated WebP(AWebP) image format.
+ Image/IO provide the WebP decoding support in iOS 14/macOS 11/tvOS 14/watchOS 7+.
+ @note Currently Image/IO seems does not supports WebP encoding, if you need WebP encoding, use the custom codec below.
+ @note If you need to support lower firmware version for WebP, you can have a try at https://github.com/SDWebImage/SDWebImageWebPCoder
+ */
+@interface SDImageAWebPCoder : SDImageIOAnimatedCoder <SDProgressiveImageCoder, SDAnimatedImageCoder>
```
Comment (kinarobin) [inline SDWebImage/Core/SDImageAWebPCoder.h:18]: Should we label like `API_AVAILABLE(ios(14.0), tvos(14.0), watchos(7.0), macos(11.0))`?

File: SDWebImage/Core/SDImageIOAnimatedCoder.m [hunk 3]
```diff
@@ -108,6 +108,32 @@ + (NSUInteger)defaultLoopCount {
...
 #pragma mark - Utils
 
++ (BOOL)canDecodeFromFormat:(SDImageFormat)format {
+    CFStringRef imageUTType = [NSData sd_UTTypeFromImageFormat:format];
+    NSArray *imageUTTypes = (__bridge_transfer NSArray *)CGImageSourceCopyTypeIdentifiers();
+    if ([imageUTTypes containsObject:(__bridge NSString *)(imageUTType)]) {
+        // Can decode from target format
+        return YES;
+    }
+    return NO;
```
Comment (kinarobin) [inline SDWebImage/Core/SDImageIOAnimatedCoder.m:118]: How about change `NSArray *imageUTTypes` to  `static  NSSet *imageUTTypes`, In this way, other coders can directly call the contain method, and each coder does not need to declare its own internal method, like:
```objc
+ (BOOL)canDecodeFromWebPFormat;
+ (BOOL)canDecodeFromHEICFormat;
+ (BOOL)canDecodeFromHEIFFormat;
```

File: SDWebImage/Core/SDImageAWebPCoder.h (hunk 2 above)
Comment (dreampiggy) [reply SDWebImage/Core/SDImageAWebPCoder.h:18]: SDImageHEIFCoder does not have this limit as well. If we add, add them together. If not, keep it.

File: SDWebImage/Core/SDImageIOAnimatedCoder.m (hunk 3 above)
Comment (dreampiggy) [reply SDWebImage/Core/SDImageIOAnimatedCoder.m:118]: It's OK with this optimization to use RAM to trade off the CPU. But the access to Set seems not performent than use static once method.

Comment (dreampiggy) [reply SDWebImage/Core/SDImageIOAnimatedCoder.m:118]: We can still keep these methods. Does this cause something issue ?

File: SDWebImage/Core/SDImageAWebPCoder.h (hunk 2 above)
Comment (kinarobin) [reply SDWebImage/Core/SDImageAWebPCoder.h:18]: How about add them together?

Comment (dreampiggy) [reply SDWebImage/Core/SDImageAWebPCoder.h:18]: OK

File: SDWebImage/Core/SDImageIOAnimatedCoder.m (hunk 3 above)
Comment (kinarobin) [reply SDWebImage/Core/SDImageIOAnimatedCoder.m:118]: I mean, It's just that you don't need to declare it in an internal file, call the unified method directly.  Thus, If you add a new coder, you don't need add the `canDecodeFromNewFormat` && `canEncodeToNewFormat`  method in internal file.

**Pull request merged**